site/docs/skylark/aspects.md

```diff
@@ -116,85 +150,248 @@ functions. They return [providers](rules.md#providers), can generate
 *    `ctx`: [`ctx`](lib/ctx.html) object that can be used to access attributes and
      generate outputs and actions.
 
-Example:
+The implementation function can access the attributes of the target rule via
+[`ctx.rule.attr`](lib/ctx.html#rule). It can examine providers that are
+provided  by the target to which it is applied (via the `target` argument).
+
+In this example the implementation doesn't return anything so `PrintProvider` is
```

Comment:
The wording of "returns anything" is confusing, as it is returning the PrintProvider

Reply:
Fixed by getting rid of PrintProvider